### PR TITLE
Bump aiofiles to 23.2.*

### PR DESCRIPTION
**Pull request by srittau**
Closes: #10551

### CODE REVIEW

**Comment by AlexWaygood** [comment]
The runtime might include the symbols in `__all__` on Windows even though it doesn't actually define them at runtime on Windows, which is a recipe for confusing our test suite

**Comment by srittau** [comment]
I've disabled stubtest for now. Alternatively we could remove `__all__` from the stubs, but hopefully this will get fixed upstream.

File: stubs/aiofiles/METADATA.toml [hunk 1]
```diff
@@ -1,6 +1,10 @@
...
+version = "23.2.*"
 upstream_repository = "https://github.com/Tinche/aiofiles"
 
 [tool.stubtest]
 # linux and darwin are equivalent
 platforms = ["linux", "win32"]
+# Currently we are in a pickle, because aiofiles.os.__all__ is incorrect at
+# runtime when running on win32. (https://github.com/Tinche/aiofiles/pull/174)
+# This means that stubtest is either correct on Windows or all other platforms.
+skip = true
```
Comment (AlexWaygood) [inline stubs/aiofiles/METADATA.toml:10]: Surely we don't need to skip stubtest entirely for the whole package. Can't we just create a `stubtest_allowlist_win32.txt` file in the `@tests` directory with a single entry, `aiofiles.os.__all__`?

Comment (srittau) [reply stubs/aiofiles/METADATA.toml:10]: I didn't know we could have architecture-specific allowlists! Will change accordingly.

Comment (AlexWaygood) [reply stubs/aiofiles/METADATA.toml:10]: One of @Avasam's innovations :)

Comment (srittau) [reply stubs/aiofiles/METADATA.toml:10]: Embarrassingly, aiofiles already has architecture-specific allowlists ... 😳

File: stubs/aiofiles/@tests/stubtest_allowlist_win32.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# At runtime, __all__ includes some items that are not available on Windows.
+# https://github.com/Tinche/aiofiles/pull/174
+aiofiles.os.__all__
```
Comment (AlexWaygood) [inline stubs/aiofiles/@tests/stubtest_allowlist_win32.txt:3]:
```suggestion
aiofiles.os.__all__
aiofiles.os.sendfile
aiofiles.os.statvfs
```

File: stubs/aiofiles/METADATA.toml (hunk 1 above)
Comment (AlexWaygood) [reply stubs/aiofiles/METADATA.toml]: > One of @Avasam's innovations :)

Oh, it was @sobolevn's PR, so he deserves most of the credit! https://github.com/python/typeshed/pull/8923 Though @Avasam pushed for the change

**Review by AlexWaygood — APPROVED** [review]
Thanks!

File: stubs/aiofiles/METADATA.toml (hunk 1 above)
Comment (Avasam) [reply stubs/aiofiles/METADATA.toml]: > Oh, it was @sobolevn's PR, so he deserves most of the credit! #8923 Though @Avasam pushed for the change

Exact :) I fixed a couple edge cases (like #9173) and updated plenty of stubs afterwards.

**Pull request merged**